### DISCUSSION TITLE
Linux共享tun

### DISCUSSION THREAD

**Discussion by quaeast** [xjasonlyu/tun2socks#197] [Q&A]
按照教程的方法成功让Linux连上socks5，但是如果我想把这个Linux作为路由器，请问如何设置路由表呢？

<img width="1268" alt="image" src="https://user-images.githubusercontent.com/35144675/205848640-21c8071d-aa0c-4e3a-8a43-944d70a0f4d1.png">

**Comment by xjasonlyu** [comment]
你直接把其他设备的网关改成这个就行了

**Reply by quaeast** [reply]
按理说应该是好使的，但我的手机这样设置之后并不好使。Linux服务器如果不调整路由表，手机这样设置是可以访问到百度的，但是根据教程输入这行命令把默认网关修改成tun之后，Linux可以联网但手机就没办法联网了：

```sh
ip route add default via 198.18.0.1 dev tun0 metric 1
```

<img width="846" alt="image" src="https://user-images.githubusercontent.com/35144675/205850586-d4a07fc9-dd52-4d26-81cc-4ac47044d547.png">
<img width="874" alt="image" src="https://user-images.githubusercontent.com/35144675/205850620-e380146f-3cd6-4eb3-b3cf-212b4f2d4c56.png">

**Reply by xjasonlyu** [reply]
手机上配置好网关后，tun2socks上有什么显示吗？

**Reply by quaeast** [reply]
大部分都是udp的dns流量，但是我感觉这个应该不是手机发送的而是Linux服务器自己发的，因为114这个dns是Linux的dns。

<img width="799" alt="image" src="https://user-images.githubusercontent.com/35144675/205852581-91b0c9e7-602a-47ef-a13b-0de9c590f5db.png">
